### PR TITLE
Create a new file, adding sub object from catalog

### PR DESCRIPTION
**Pull request by shivampdesai**
- fixed bug so that a child gets removed from it's parent connections list when it is dragged outside of it
- changed resizing so we only resize parent when needed (child is sticking out of parent)

### CODE REVIEW

File: sym_object.py [hunk 1]
```diff
@@ -567,3 +577,38 @@ def updateName(self, newName):
...
+        child.z = self.z + 1
+        if not child.name in self.connected_objects:
+            self.connected_objects.append(child.name)
+
+        for object in self.state.sym_objects.values():
+            object.setZValue(object.z)
+
+    def resizeParent(self, child):
+        '''reduce the size of the parent if it has one child'''
+        self.removeFromGroup(self.rect)
```
Comment (afarooqui98) [inline sym_object.py:595]: these happen a couple of times in UI resizing, intitialization, etc. i know this is probably a derived QT function but if the items being modified are the same, can we move these batch updates to its own function? could be wrong

File: gui_views/catalog_view.py [hunk 2]
```diff
@@ -53,18 +53,57 @@ def createSymObject(self, item):
...
+        new_object = \
             self.state.scene.addObjectToScene("component", item.text(0), name)
-        self.state.selected_sym_objects[0].instance_params = \
+        new_object.instance_params = \
             copy.deepcopy(self.catalog[item.parent().text(0)][item.text(0)]['params'])
-        self.state.selected_sym_objects[0].instance_ports = \
+        new_object.instance_ports = \
             copy.deepcopy(self.catalog[item.parent().text(0)][item.text(0)]['ports'])
-        self.state.selected_sym_objects[0].initPorts()
+        new_object.initPorts()
```
Comment (afarooqui98) [inline gui_views/catalog_view.py:69]: overwriting rohit's changes?

File: sym_object.py (hunk 1 above)
Comment (afarooqui98) [reply sym_object.py:595]: Sorry, in this case im referring to the successive removeFromGroup calls

File: gui_views/catalog_view.py (hunk 2 above)
Comment (shivampdesai) [reply gui_views/catalog_view.py:69]: Yes, spoke with Rohit and we agreed that we don't want to highlight objects on load by default - only if we are adding a child through the catalog (which we do later on in the function).

File: sym_object.py (hunk 1 above)
Comment (shivampdesai) [reply sym_object.py:595]: Good point, will refactor those into its own function.

**Pull request merged**